format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/PathTrackingDelimitedInputFormat.java

```diff
@@ -80,18 +148,27 @@ public NullWritable getCurrentKey() {
 
         StructuredRecord.Builder builder = StructuredRecord.builder(schema);
         Iterator<Schema.Field> fields = schema.getFields().iterator();
-        Iterable<String> splitDelimitedString = Splitter.on(delimiter).split(delimitedString);
+        Iterable<String> splits;
+        if (!enableQuotesValue) {
+          splits = Splitter.on(delimiter).split(delimitedString);
+        } else {
+          splits = splitQuotesString(delimitedString, delimiter);
+        }
 
-        for (String part : splitDelimitedString) {
+        int numSchemaFields = schema.getFields().size();
+        int numDataFields = 0;
```

Comment:
this should be moved back to where it was before. It is only done in the error scenario in order to prevent looping an extra time for each line.